hikari/impl/special_endpoints.py

```diff
@@ -1682,29 +1674,96 @@ def build(self) -> typing.MutableMapping[str, typing.Any]:
         return data
 
 
-@attr_extensions.with_copy
-@attr.define(kw_only=True, weakref_slot=False)
-class TextSelectMenuBuilder(
-    SelectMenuBuilder[_ContainerProtoT], special_endpoints.TextSelectMenuBuilder[_ContainerProtoT]
-):
+@attr.define(init=False, weakref_slot=False)
+class TextSelectMenuBuilder(SelectMenuBuilder, special_endpoints.TextSelectMenuBuilder[_ParentT]):
     """Builder class for text select menus."""
 
-    _type: typing.Literal[component_models.ComponentType.TEXT_SELECT_MENU] = attr.field(
-        default=component_models.ComponentType.TEXT_SELECT_MENU, init=False
-    )
-    # Any has to be used here as we can't access Self type in this context
-    _options: typing.List[special_endpoints.SelectOptionBuilder[typing.Any]] = attr.field(alias="options", factory=list)
+    _options: typing.List[special_endpoints.SelectOptionBuilder] = attr.field()
+    _parent: typing.Optional[_ParentT] = attr.field()
+    _type: typing.Literal[component_models.ComponentType.TEXT_SELECT_MENU] = attr.field()
+
+    if not typing.TYPE_CHECKING:
+        # This will not work with the generated for attrs copy methods.
+        __copy__ = None
+        __deepcopy__ = None
+
+    @typing.overload
+    def __init__(
+        self,
+        *,
+        custom_id: str,
+        parent: _ParentT,
+        options: typing.Sequence[special_endpoints.SelectOptionBuilder] = (),
+        placeholder: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        min_values: int = 0,
+        max_values: int = 1,
+        is_disabled: bool = False,
+    ) -> None:
+        ...
+
+    @typing.overload
+    def __init__(
+        self: TextSelectMenuBuilder[typing.NoReturn],
+        *,
+        custom_id: str,
+        options: typing.Sequence[special_endpoints.SelectOptionBuilder] = (),
+        placeholder: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        min_values: int = 0,
+        max_values: int = 1,
+        is_disabled: bool = False,
+    ) -> None:
+        ...
+
+    def __init__(
+        self,
+        *,
+        custom_id: str,
+        parent: typing.Optional[_ParentT] = None,
+        options: typing.Sequence[special_endpoints.SelectOptionBuilder] = (),
+        placeholder: undefined.UndefinedOr[str] = undefined.UNDEFINED,
+        min_values: int = 0,
+        max_values: int = 1,
+        is_disabled: bool = False,
+    ) -> None:
+        super().__init__(
+            type=component_models.ComponentType.TEXT_SELECT_MENU,
+            custom_id=custom_id,
+            placeholder=placeholder,
+            min_values=min_values,
+            max_values=max_values,
+            is_disabled=is_disabled,
+        )
+        self._options = list(options)
+        self._parent = parent
+
+    @property
+    def parent(self) -> _ParentT:
+        if self._parent is None:
+            raise RuntimeError("This menu has no parent")
```

Comment:
Should we allow text select menu builders without a parent?

Reply:
How else would you pass them to `TextSelectMenuBuilder.__init__`